src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -37,6 +37,25 @@ public abstract class ExecutionNode
         /// </summary>
         public int? IndexInParentNode { get; protected set; }
 
+        //todo: make public
+        internal IGraphType ResolvedType
+        {
+            get
+            {
+                if (IndexInParentNode.HasValue)
+                {
+                    var t = FieldDefinition.ResolvedType;
+                    if (t is NonNullGraphType nonNullGraphType)
+                        t = nonNullGraphType.ResolvedType;
+                    return ((ListGraphType)t).ResolvedType;
```

Comment:
```suggestion
                   return FieldDefinition.ResolvedType.GetNamedType();
```

Reply:
But why you unwrap `NonNullGraphType` ? I see non null check on line https://github.com/graphql-dotnet/graphql-dotnet/pull/2064/files#diff-df2b6c74045af40593068c2612253adc21462ab45546c45ca3a27bd29fd83ab1R369

Reply:
`FieldDefinition.ResolvedType` is guaranteed to be a list graph type.  I need to unwrap the list only, not the nullability of the inside type.  In order to unwrap the list, I first need to check if it's a non-null list.  This code specifically does not unwrap the nullability of the interior graph type, as that is its entire purpose.  This is also why `GetNamedType()` does not work.

Reply:
Note that the `ExecutionNode.GraphType` property already equals `FieldDefinition?.ResolvedType.GetNamedType()`, I believe.  (It's set from the constructor, but that's basically what it contains.)

Reply:
> FieldDefinition.ResolvedType is guaranteed to be a list graph type in this context. I need to unwrap the list only, not the nullability of the inside type. In order to unwrap the list, I first need to check if it's a non-null list. This code specifically does not unwrap the nullability of the interior graph type, as that is its entire purpose. This is also why GetNamedType() does not work.

I highly recommend adding comments about this. It wasn't obvious to me.

Reply:
> Note that the ExecutionNode.GraphType property already equals FieldDefinition?.ResolvedType.GetNamedType(), I believe. (It's set from the constructor, but that's basically what it contains.)

I looked at the code and I had suspicions that we can make mistakes and pass wrapped types sometime, and sometimes not. I suggest to add type checks into `ExecutionNode` and its desccendants ctors to enforce rules and make rules explicit.

Reply:
Done. Please review

Reply:
Added checks for `ExecutionNode` ctors in PR #2069